Zend/zend_language_scanner.l

```diff
@@ -1563,6 +1658,15 @@ NEWLINE ("\r"|"\n"|"\r\n")
 
 
 <ST_IN_SCRIPTING>"}" {
+	if(in_class) {
+		if(--in_class == 1) {
+			in_class = 0;	
+		}
+		if(in_use_list) {
+			in_use_list = 0;
+			in_trait_conflict = 0;
+		}
+	}
```

Comment:
Ok thinking a bit more, there is other possibility:

If somebody used `BEGIN()` to discard `ST_DOUBLE_QUOTES`, `ST_BACKQUOTE` or `ST_HEREDOC` states instead of `yy_pop_state()` this would break the state trail and the previously pushed states will get lost. It should be fixable. But if it's not fixable... game over.